File: Clases 2018/Clases-C#/ClasePractica7_2018 (abm usuarios en capas)/C#_TPS_BugTracker/TPS_BugTracker/BusinessLayer/Services/UsuarioService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.DataLayer.DAOs;

namespace TPS_BugTracker.BusinessLayer.Services
{
    public class UsuarioService
    {
        private UsuarioDao oUsuarioDao;

        public UsuarioService()
        {
            oUsuarioDao = new UsuarioDao();
        }

        public Usuario validarUsuario(string nombre, string password)
        {
            return oUsuarioDao.getUserByNamePass(nombre, password);
        }

        public Usuario validarNombreUsuario(string nombre)
        {
            return oUsuarioDao.getUserByName(nombre);
        }

        public IList<Usuario> consultarUsuariosConFiltros(List<object> @params)
        {
            return oUsuarioDao.getByFilters(@params);
        }

        public IList<Usuario> consultarUsuarios()
        {
            return oUsuarioDao.getAll();
        }


        public IList<Usuario> consultarDesarrolladores()
        {
            List<object> @params = new List<object>() { 3, null }; // Id_perfil: 3, usuarios desarrolladores
            return oUsuarioDao.getByFilters(@params);
        }

        public bool crearUsuario(Usuario oUsuario)
        {
            return oUsuarioDao.create(oUsuario);
        }

        public bool actualizarUsuario(Usuario oUsuario)
        {
            return oUsuarioDao.update(oUsuario);
        }

        public bool modificarEstadoUsuario(Usuario oUsuario)
        {
            return oUsuarioDao.update(oUsuario);
        }
    }

}

File: PAV 1-TPI - CASOS DE USO version 1/Resueltos/Usuarios+SeleccionActBugs/App_BugTracker resuelto/App_BugTracker/BusinessLayer/Services/UsuarioService.vb
Public Class UsuarioService
    Private oUsuarioDao As UsuarioDao

    Public Sub New()
        oUsuarioDao = New UsuarioDao
    End Sub

    Public Function validarUsuario(ByVal nombre As String, ByVal password As String) As Usuario
        Return oUsuarioDao.getUserByNamePass(nombre, password)
    End Function

    Public Function validarNombreUsuario(ByVal nombre As String) As Usuario
        Return oUsuarioDao.getUserByName(nombre)
    End Function

    Public Function consultarUsuariosConFiltros(ByVal params As Object()) As List(Of Usuario)
        Return oUsuarioDao.getByFilters(params)
    End Function

    Public Function consultarUsuarios() As List(Of Usuario)
        Return oUsuarioDao.getAll()
    End Function


    Public Function consultarDesarrolladores() As List(Of Usuario)
        Dim params As Object() = {3, Nothing} 'Id_perfil: 3, usuarios desarrolladores
        Return oUsuarioDao.getByFilters(params)
    End Function

    Public Function crearUsuario(ByVal oUsuario As Usuario) As Boolean
        Return oUsuarioDao.create(oUsuario)
    End Function

    Public Function actualizarUsuario(ByVal oUsuario As Usuario) As Boolean
        Return oUsuarioDao.update(oUsuario)
    End Function

    Public Function modificarEstadoUsuario(ByVal oUsuario As Usuario) As Boolean
        Return oUsuarioDao.update(oUsuario)
    End Function

End Class
